File: HAI912I - Développement mobile avancé, IoT et embarqué/Flutter/TP0/Product.dart
class Product {

  static int countID = 0;  
   // field 
   int? id;  
   String? name;  
   int? price;  
   String? date;  


   Product(String name, int price, String date){
    this.id = countID;
    ++countID;
    this.name = name;
    this.price = price;
    this.date = date;
   }
   
   // function 
   void Afficher() { 
      print("ID = $id"); 
      print("NOM = $name");
      print("PRIX= $price");
      print("DATE = $date" );
   } 
}

File: HAI913I - Évolution et restructuration des logiciels/Cours3/TP/TP3_rustructuration_rendu/TP3-Logging/src/main/java/me/sanchez/logging/Product/Product.java
package me.sanchez.logging.Product;
import java.util.Objects;
public class Product {
    private Long id;

    private String nom;

    private Double prix;

    private String expDate;

    private static Long countID = 0L;

    public Product(String nom, Double prix, String expDate) {
        this.id = countID;
        ++countID;
        this.nom = nom;
        this.prix = prix;
        this.expDate = expDate;
    }

    @Override
    public String toString() {
        return "(" + id + ") " + nom + " - "+expDate+"- : " + prix.toString();
    }

    public Long getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public Double getPrix() {
        return prix;
    }

    public void setPrix(Double prix) {
        this.prix = prix;
    }

    public String getExpDate() {
        return expDate;
    }

    public void setExpDate(String expDate) {
        this.expDate = expDate;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;

        if ((o == null) || (getClass() != o.getClass()))
            return false;

        Product product = ((Product) (o));
        return (Objects.equals(nom, product.nom) && Objects.equals(prix, product.prix)) && Objects.equals(expDate, product.expDate);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nom, prix, expDate);
    }
}